specs/Standards/Tokens/NonFungibleToken.md

```diff
@@ -0,0 +1,196 @@
+
+There are a few concepts in the scenarios above:
+
+- **Total supply**. The total number of tokens in circulation.
+- **Token owner**. An account ID that owns one or more tokens.
+- **Transfer**. Action that moves some amount from one account to another account.
+- **Escrow**. A different account from the balance owner who has permission to one or more tokens.
+- **Access**. The specific token ID that another account has access to.
+
+### Simple transfer
+
+#### **Assumptions**
+
+- the Corgi nft contract is `corgi`
+- Alice's account is `alice`
+- Jeraldo's account is `jerry`
+- The NFT contract has been initialized with a nonzero token supply
+- There exists a token with the ID of `3`
+
+#### **High-level**
+
+Alice needs to issue one transaction to the Corgi NFT contract to transfer one corgi token to Jeraldo.
+
+#### **Technical calls**
+
+1. `alice` calls `corgi::transfer({"new_owner_id":"jerry", "token_id":3})`
+
+### Token transfer through a third party escrow
+
+Alice wants to transfer one Corgi NFT through a third party escrow to Jeraldo in exchange for one Sausage NFT.
+
+#### **Assumptions**
+
+- the Corgi nft contract is `corgi`
+- the Sausage nft contract is `sausage`
+- Alice's account is `alice`
+- Jeraldo's account is `jerry`
+- The Escrow contract is `escrow`
+- The NFT contract has been initialized with a nonzero token supply
+- There exists a Corgi token with the ID of `3` and a Sausage token with the ID of `5`
+- The Escrow contract manages how the transfer is facilitated and guarantees requirements are met for transfer
+
+#### **High-level**
+
+Both Alice and Jerry will issue asynchronous transactions to their respective contracts, `corgi` and `sausage` to grant access to the token ID they would like to trade. `escrow` will call the `sasuage` token contract asynchrounously to transfer the Sausage token to Alice.  `escrow` will also call the `corgi` contract to asynchornously transfer the Corgi token to Jerry.
+
+- If both of the `transferFrom` calls succeed, then Alice will now own one Sausage token and Jerry will own one Corgi token.
+- If one or both of the `transferFrom` calls fail, then nothing will happen and `escrow` should trigger a notification to both Alice and Jerry that their transfers were unsuccessful along with the reason.
+
+#### **Technical calls**
+
+1. `alice` makes an async call to `corgi::grantAccess({"escrow_account_id":"escrow", "token_id":3})`
+2. `jerry`  makes an async call to ``sausage::grantAccess({"escrow_account_id":"escrow", "token_id":5})`
+3. In one Promise
+    1. `escrow` calls `sausage::transferFrom({"owner_id":"jerry", "new_owner_id:"alice", "token_id": 5})`
+        1. attaches callback `escrow::onTransfer({"owner_id":"jerry", "token_contract":"sausage", "token_id": 5})`
+    2. `escrow` calls `corgi::transferFrom({"owner_id":"alice", "new_owner_id:"jerry", "token_id": 3})`
+        1. attaches callback `escrow::onTransfer({"owner_id":"alice", "token_contract:"corgi", "token_id": 3})`
```

Comment:
This is wrong example, because if one promise failed, the other succeeds and escrow can't do anything with the successful transfer. This would mean the owner has both assets. You need to adjust this example to follow similar logic to https://nomicon.io/Standards/Tokens/FungibleToken.html#multi-token-swap-on-dex

Basically escrow first takes both assets into ownership. This way escrow contract can be sure that it holds both assets, then it can swap them respectively.